lib/statsd/instrument/metric.rb

```diff
@@ -61,6 +61,9 @@ def initialize(options = {})
     @value       = options[:value] || default_value
     @sample_rate = options[:sample_rate] || StatsD.default_sample_rate
     @tags        = StatsD::Instrument::Metric.normalize_tags(options[:tags])
+    if StatsD.default_tags
+      @tags        = Array(@tags) | StatsD::Instrument::Metric.normalize_tags(StatsD.default_tags)
```

Comment:
For performance reasons, can we normalize `default_tags` when you assign them, rather than for every metric that gets emitted?

Reply:
One question is how we want to handle duplicate tags. How do we want to handle a situation when we have a tag that conflicts with a default tag (e.g. `foo:bar` and `foo:baz`). My instinct tells me that the value in `tags` should overwrite the value in `default_tags`.

We cannot really check for duplicates here for performance reasons. However, we can influence the ordering. Can you check what value will be used if you have two tags with the same key: the first or the last one? Based on that, we can determine whether we want to do `tags | default_tags` or `default_tags | tags`.

Reply:
The way this is currently implemented does not handle tags with duplicated keys and the resulting list will include both unless their "key:value" are the same.

```ruby
options[:tags] = {"foo" => "bar"}
StatsD.default_tags = {"foo" => "baz"}

@tags = StatsD::Instrument::Metric.normalize_tags(options[:tags])
=> ["foo:bar"]
if StatsD.default_tags
  @tags        = Array(@tags) | StatsD::Instrument::Metric.normalize_tags(StatsD.default_tags)
end
=> ["foo:bar", "foo:baz"]
```

Reply:
So my question is: how does Datadog interpret this?

Reply:
It is interpreted as separate tags and ordering is irrelevant, I think. I tested that both can be used as any other tags, for instance, to narrow down searches.

```ruby
StatsD.increment("fmaia_test", tags: ["foo:bar", "foo:baz"])
```

![image](https://user-images.githubusercontent.com/663329/64989047-b7b27580-d89a-11e9-9e29-6e5468bf0a41.png)

![image](https://user-images.githubusercontent.com/663329/64989106-d1ec5380-d89a-11e9-91cb-b96f7d471879.png)

Reply:
I guess this is acceptable

Reply:
Sure. I'll add this to the PR.

Reply:
@fabiormoura is that DataDog?

Reply:
> @fabiormoura is that DataDog?

It is @scalp42